plugins/backend-trace-forwarder/src/Cardano/BM/Backend/TraceForwarder.lhs

```diff
@@ -161,3 +181,94 @@ data TraceForwarderBackendFailure
 instance Exception TraceForwarderBackendFailure
 
 \end{code}
+
+\subsubsection{Asynchronously reading log items from the queue and sending them to an acceptor.}
+\begin{code}
+spawnDispatcher :: ToJSON a => TraceForwarderMVar a -> IO (Async.Async ())
+spawnDispatcher tfMVar = Async.async $ processQueue
+ where
+  processQueue :: IO ()
+  processQueue = do
+    currentTF <- readMVar tfMVar
+    -- Read the next log item from the queue. If the queue is still empty -
+    -- blocking and waiting for the next log item.
+    nextItem <- atomically $ TBQ.readTBQueue (tfQueue currentTF)
+    -- Try to write it to the handle. If there's a problem with connection,
+    -- this thread will initiate re-establishing of the connection and
+    -- will wait until it's established.
+    sendItem tfMVar nextItem
+    -- Continue...
+    processQueue
+
+-- | Try to send log item to the handle.
+sendItem :: ToJSON a => TraceForwarderMVar a -> LogObject a -> IO ()
+sendItem tfMVar lo = do
+  tfHandle <$> readMVar tfMVar >>= \case
+    Nothing -> do
+      -- There's no handle, initiate the connection.
+      establishConnection 1 1 tfMVar
+      -- Connection is re-established, try to send log item.
+      sendItem tfMVar lo
+    Just h ->
+      try (BSC.hPutStrLn h $! encodeUtf8 (hostname . loMeta $ lo)) >>= \case
```

Comment:
why not create a "where" clause for the `encodedHostname`?

Reply:
Done.